SpatialGDK/Source/SpatialGDK/Private/Interop/OwnershipCompletenessHandler.cpp

```diff
@@ -45,29 +45,35 @@ bool FOwnershipCompletenessHandler::ShouldHaveOwnerOnlyComponents(Worker_EntityI
 
 void FOwnershipCompletenessHandler::AddPlayerEntity(Worker_EntityId EntityId)
 {
-	PlayerOwnedEntities.Add(EntityId);
-
-	for (FSubView* SubViewToRefresh : SubViewsToRefresh)
-	{
-		SubViewToRefresh->Refresh();
-	}
+	bool bIsAlreadyInSet = false;
+	PlayerOwnedEntities.Add(EntityId, &bIsAlreadyInSet);
+	bRequiresRefresh = bRequiresRefresh || !bIsAlreadyInSet;
```

Comment:
I don't know if I like this version better, less arithmetic 
```suggestion
	if (!bIsAlreadyInSet)
	{
		bRequiresRefresh = true;
	}
```

Reply:
Hah, I always steer away from branching logic since my shader days, but I agree this is simpler 👍